python-runner/server.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python3
+"""
+Python execution server for Metabase.
+Runs Python scripts in isolated processes with resource limits and monitoring.
+"""
+
+import os
+import sys
+import time
+import signal
+import resource
+import threading
+import subprocess
+from pathlib import Path
+from queue import Queue, Empty
+from datetime import datetime, timedelta
+from typing import Optional, Dict, Any
+
+from flask import Flask, request, jsonify
+from werkzeug.exceptions import HTTPException
+
+DEFAULT_TIMEOUT = 30  # seconds
+MAX_MEMORY_MB = 1024  # Increased for pandas/numpy
+MAX_CPU_TIME_SECONDS = 60
+MAX_QUEUE_SIZE = 10
```

Comment:
Thought about just straight up failing if there's a current process Metabase tries to call without a Cluster lock.

This queue didn't add that much complexity though, and nice to remove the external co-ordination.